include/cooperative_perception/angle.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COOPERATIVE_PERCEPTION_ANGLE_HPP
+#define COOPERATIVE_PERCEPTION_ANGLE_HPP
+
+#include <complex>
+#include <units.h>
+#include <boost/math/constants/constants.hpp>
+#include "cooperative_perception/utils.hpp"
+
+namespace cooperative_perception
+{
+/**
+ * @brief Angle type for representing and manipulating angles
+ */
+class Angle
+{
+private:
+  std::complex<double> value_{};
+
+public:
+  Angle(units::angle::radian_t angle_val) noexcept : value_{ units::math::cos(angle_val), units::math::sin(angle_val) }
+  {
+  }
+
+  auto set_angle(units::angle::radian_t angle_val) noexcept -> void
+  {
+    value_.real(units::math::cos(angle_val));
+    value_.imag(units::math::sin(angle_val));
+  }
+
+  auto get_angle() const noexcept -> units::angle::radian_t
+  {
+    auto phase_angle = std::arg(value_);
+    if (phase_angle < 0.0)
+    {
+      phase_angle += 2.0 * boost::math::double_constants::pi;
+    }
+    return units::angle::radian_t{ phase_angle };
+  }
+
+  friend auto operator==(const Angle& lhs, const Angle& rhs) -> bool
+  {
+    return lhs.value_ == rhs.value_;
+  }
+};
+
+inline auto operator+=(Angle& lhs, const Angle& rhs) noexcept -> Angle&
+{
+  auto angle_sum = lhs.get_angle() + rhs.get_angle();
+  lhs.set_angle(angle_sum);
```

Comment:
Not a huge issue but this is a very computationally expensive way of adding what are ultimately two numbers (multiple trig calls). I assume it's required due to the storage as a complex number but wanted to double check if there's any faster way to do it.

Reply:
I tried alternative implementations that dealt with the `std::complex` variables directly to try and avoid the getters and setters, but the [benchmarks](https://quick-bench.com/q/XdXSr_V86D7sd12kg-xv8GExBus) did not show any improvement. Some implementations were worse.

The interface isn't going to change, so I suggest we stick with this implementation for now and revisit later if we need to.

Reply:
Sounds good thanks for checking. That's a cool little benchmarking tool!